drivers/lsm6ds3/lsm6ds3.c

```diff
@@ -0,0 +1,298 @@
+          | LSM6DS3_CTRL9_XL_Z_EN);    /**< enable z-axis on accelerometer */
+    i2c_write_reg(dev->i2c, dev->addr, LSM6DS3_CTRL9_XL, tmp);
+
+    tmp = ( acc_odr                    /**< accelerometer ODR */
+          | acc_scale);                /**< accelerometer FS */
+    i2c_write_reg(dev->i2c, dev->addr, LSM6DS3_CTRL1_XL, tmp);
+
+    tmp = ( LSM6DS3_CTRL10_G_X_EN      /**< enable x-axis on gyroscope */
+          | LSM6DS3_CTRL10_G_Y_EN      /**< enable y-axis on gyroscope */
+          | LSM6DS3_CTRL10_G_Z_EN);    /**< enable z-axis on gyroscope */
+    i2c_write_reg(dev->i2c, dev->addr, LSM6DS3_CTRL10_C, tmp);
+
+    tmp = ( gyr_odr                    /**< gyroscope ODR */
+          | gyr_scale );               /**< gyroscope FS */
+    i2c_write_reg(dev->i2c, dev->addr, LSM6DS3_CTRL2_G, tmp);
+
+    i2c_release(dev->i2c);
+
+    return 0;
+}
+
+void lsm6ds3_read_acc(lsm6ds3_t *dev, lsm6ds3_3d_data_t *data)
+{
+    char tmp;
+    float sensitivity = 0;
+    int16_t raw[3];    /**< raw[0] := X-, raw[1] := Y-, raw[2] := Z-axis */
+
+    _acc_raw(dev, raw);
+
+    i2c_acquire(dev->i2c);
+    i2c_read_reg(dev->i2c, dev->addr, LSM6DS3_CTRL1_XL, &tmp);
+    i2c_release(dev->i2c);
+
+    tmp &= LSM6DS3_MASK_CTRL1_XL_FS;
+
+    switch (tmp) {
+
+        case LSM6DS3_MASK_CTRL1_XL_FS_2G:
+            sensitivity = 0.061;
+            break;
+
+        case LSM6DS3_MASK_CTRL1_XL_FS_4G:
+            sensitivity = 0.122;
+            break;
+
+        case LSM6DS3_MASK_CTRL1_XL_FS_8G:
+            sensitivity = 0.244;
+            break;
+
+        case LSM6DS3_MASK_CTRL1_XL_FS_16G:
+            sensitivity = 0.488;
+            break;
+    }
+
+    data->x_axis = raw[0] * sensitivity;
+    data->y_axis = raw[1] * sensitivity;
+    data->z_axis = raw[2] * sensitivity;
+}
+
+void lsm6ds3_read_gyr(lsm6ds3_t *dev, lsm6ds3_3d_data_t *data)
+{
+    char tmp;
+    float sensitivity = 0;
+    int16_t raw[3];    /**< raw[0] := X-, raw[1] := Y-, raw[2] := Z-axis */
+
+    _gyr_raw(dev, raw);
+
+    i2c_acquire(dev->i2c);
+    i2c_read_reg(dev->i2c, dev->addr, LSM6DS3_CTRL2_G, &tmp);
+    i2c_release(dev->i2c);
+
+    tmp &= LSM6DS3_MASK_CTRL2_G_FS;
+
+    switch (tmp) {
+
+        case LSM6DS3_MASK_CTRL2_G_FS_125DPS:
+            sensitivity = 4.375;
+            break;
+
+        case LSM6DS3_MASK_CTRL2_G_FS_245DPS:
+            sensitivity = 8.75;
+            break;
+
+        case LSM6DS3_MASK_CTRL2_G_FS_500DPS:
+            sensitivity = 17.50;
+            break;
+
+        case LSM6DS3_MASK_CTRL2_G_FS_1000DPS:
+            sensitivity = 35;
+            break;
+
+        case LSM6DS3_MASK_CTRL2_G_FS_2000DPS:
+            sensitivity = 7;
+            break;
+    }
+
+    data->x_axis = raw[0] * sensitivity / GYR_DIVIDER;
+    data->y_axis = raw[1] * sensitivity / GYR_DIVIDER;
+    data->z_axis = raw[2] * sensitivity / GYR_DIVIDER;
+}
+
+void lsm6ds3_read_temp(lsm6ds3_t *dev, int16_t *data)
```

Comment:
should be `value` instead of `data` here or `data` below insteal of `value`, otherwise it doesn't build